wrpclient/wrpclient.go

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright 2022 Comcast Cable Communications Management, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package wrpclient
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/xmidt-org/httpaux/erraux"
+	"github.com/xmidt-org/wrp-go/v3"
+)
+
+var (
+	errEncoding              = errors.New("encoding error")
+	errCreateRequest         = errors.New("http request creation error")
+	errHTTPTransaction       = errors.New("http transaction error")
+	errDecoding              = errors.New("decoding response error")
+	errNonSuccessfulResponse = errors.New("non-200 response")
+	errInvalidRequestFormat  = errors.New("invalid client RequestFormat")
+	errInvalidURL            = errors.New("invalid client URL")
+)
+
+type HTTPClient interface {
+	Do(*http.Request) (*http.Response, error)
+}
+
+type Client struct {
+
+	// URL is the full location for the serverside wrp endpoint.
+	// If unset, use talaria's URI at localhost, which is the port used in talaria's docker image
+	url string
+
+	// RequestFormat is the wrp Format to use for all requests, which specifies the wrp.Encoder.
+	// If unset, defaults to JSON, which is what the wrp package defaults to.
+	requestFormat wrp.Format
+
+	// If unset, defaults to net/http.DefaultClient
+	httpClient HTTPClient
+}
+
+func New(u string, r wrp.Format, h HTTPClient) (*Client, error) {
```

Comment:
```suggestion
func New(reqURL string, reqFormat wrp.Format, httpClient HTTPClient) (*Client, error) {
```
Since this function is exported, the parameter names being self documenting helps any callers of it.